src/Resources/views/BlockAdmin/select_type.html.twig

```diff
@@ -21,21 +21,26 @@ file that was distributed with this source code.
             </h3>
         </div>
         <div class="box-body">
-            {% for code, service in services %}
+            {% for code, service in services if service.metadata is defined or service.blockMetadata is defined %}
```

Comment:
`for .. if` syntax is deprecated. We should try to use filters instead.

Reply:
No idea how to write this without creating a new twig extension or a nested if statement inside the loop

Reply:
```suggestion
            {% for code, service in services|filter(service => service.metadata is defined or service.blockMetadata is defined) %}
```
But please note that `filter` filter was added at Twig 2.10: https://twig.symfony.com/doc/2.x/filters/filter.html.

Reply:
So we should keep it as it is, because we support twig 1/2. We could remove the `if` when dropping support for twig 1

Reply:
I think we can drop the support for Twig 1.x in a minor version, as [we did before in other bundles](https://github.com/sonata-project/SonataAdminBundle/commit/7619f20f25fea2432784c490539557a68424cf96#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780R55).